### PR TITLE
[VPAT:Forum] Tab navigation forum fixes

### PR DESCRIPTION
**Pull request by jchulton**
Addresses most of the issues from issue #3539. On the main page of the discussion forum, there were some bugs with tab navigation. 

The first was the incorrect ordering of More and Create Thread (tabbing when Filter is focused goes to Create Thread, instead of More). This occurred because the two buttons were centered to the right, so the rightmost button (Create Thread) is actually the first button. I fixed this by using fixed positioning (setting each button to appear at an exact part of their parent div) instead of floating them to the right. This will be more annoying if their are a variable amount of buttons in the div. 

I also fixed several of the buttons being skipped by including the tabindex field to their divs and setting it to 0. More of a quick, case-by-case fix, but it should work for now.

It looks like now focus starts on the currently selected thread in the thread list. Is this the intended behavior or should nothing initially have focus?

### CODE REVIEW

**Comment by bmcutler** [comment]
@jchulton please resolve conflicts

File: site/app/templates/forum/CreatePost.twig [hunk 1]
```diff
@@ -41,38 +41,38 @@
...
 
     {% if includeReply and (userGroup <= 3 or post.author_user_id == current_user) and first and thread_resolve_state == -1 %}
-        <a class="btn btn-default btn-sm post_button_color text-decoration-none" onClick="changeThreadStatus({{ post.thread_id }})" title="Mark thread as resolved">Mark as resolved</a>
+        <a class="btn btn-default btn-sm post_button_color text-decoration-none" tabindex = "0" onClick="changeThreadStatus({{ post.thread_id }})" title="Mark thread as resolved">Mark as resolved</a>
     {% endif %}
 
     <span class="post-action-container">
 
         {% if userGroup <= 2 and post.author_user_id != current_user %}
-            <a class="post-email-toggle" onClick='$(this).next().toggle();' title="Show/Hide email address" aria-label="Show/Hide email address"><i class="fas fa-envelope"></i></a>
```
Comment (elihschiff) [inline site/app/templates/forum/CreatePost.twig]: Do not remove the aria-label

File: site/app/templates/forum/CreatePost.twig [hunk 2]
```diff
@@ -41,38 +41,38 @@
...
+            <a class="post-email-toggle" tabindex = "0" onClick='$(this).next().toggle();' title="Show/Hide email address"><i class="fas fa-envelope" aria-hidden="true"></i></a>
             <a href="mailto:{{ author_email }}" style="display: none;">{{ author_email }}</a>
         {% endif %}
 
         {% if userGroup <= 2 %}
-            <a class="post-user-info" onClick='changeName(this.parentNode, {{ post_user_info.info_name }}, {{ post_user_info.visible_user_json}}, {{ post_user_info.jscriptAnonFix}})' title="Show full user information" aria-label="Show full user information"><i class="fas fa-eye"></i></a>
```
Comment (elihschiff) [inline site/app/templates/forum/CreatePost.twig]: Do not remove the aria-label

File: site/app/templates/forum/CreatePost.twig [hunk 3]
```diff
@@ -41,38 +41,38 @@
...
+            <a class="post-user-info" tabindex = "0" onClick='changeName(this.parentNode, {{ post_user_info.info_name }}, {{ post_user_info.visible_user_json}}, {{ post_user_info.jscriptAnonFix}})' title="Show full user information"><i class="fas fa-eye" aria-hidden="true"></i></a>
         {% endif %}
 
         {% if userGroup <= 3 or post.author_user_id == current_user %}
             {% if not (isThreadLocked != 1 or userAccessFullGrading == true) %}
                 {#pass#}
             {% else %}
-                <a class="post_button delete-post-button" onClick="deletePostToggle({{ post_buttons.delete.ud_toggle_status }}, {{ post.thread_id }}, {{ post.id }}, '{{ post.author_user_id }}', '{{ post_date }}', '{{ csrf_token }}' )" title="{{ post_buttons.delete.ud_button_title }}" aria-label="{{ post_buttons.delete.ud_button_title }}"><i class="fa {{ post_buttons.delete.ud_button_icon }}"></i></a>
```
Comment (elihschiff) [inline site/app/templates/forum/CreatePost.twig]: Do not remove the aria-label

File: site/app/templates/forum/CreatePost.twig [hunk 4]
```diff
@@ -41,38 +41,38 @@
...
-                <a class="post_button edit-post-button" onClick="editPost({{ post.id }}, {{ post.thread_id }}, {{ post_buttons.edit.shouldEditThread }}, {{ render_markdown?1:0 }}, '{{ csrf_token }}')" title="{{ post_buttons.edit.edit_button_title }}" aria-label="{{ post_buttons.edit.edit_button_title }}"><i class="fas fa-edit"></i></a>
```
Comment (elihschiff) [inline site/app/templates/forum/CreatePost.twig]: Do not remove the aria-label

File: site/app/templates/forum/CreatePost.twig [hunk 5]
```diff
@@ -41,38 +41,38 @@
...
+                <a class="post_button delete-post-button" tabindex = "0" onClick="deletePostToggle({{ post_buttons.delete.ud_toggle_status }}, {{ post.thread_id }}, {{ post.id }}, '{{ post.author_user_id }}', '{{ post_date }}', '{{ csrf_token }}' )" title="{{ post_buttons.delete.ud_button_title }}"><i class="fa {{ post_buttons.delete.ud_button_icon }}" aria-hidden="true"></i></a>
+                <a class="post_button edit-post-button" tabindex = "0" onClick="editPost({{ post.id }}, {{ post.thread_id }}, {{ post_buttons.edit.shouldEditThread }}, {{ render_markdown?1:0 }}, '{{ csrf_token }}')" title="{{ post_buttons.edit.edit_button_title }}"><i class="fas fa-edit" aria-hidden="true"></i></a>
```
Comment (elihschiff) [inline site/app/templates/forum/CreatePost.twig]: You should remove this aria-hidden

File: site/app/templates/forum/CreatePost.twig [hunk 6]
```diff
@@ -41,38 +41,38 @@
...
                 {#pass#}
             {% else %}
-                <a class="post_button delete-post-button" onClick="deletePostToggle({{ post_buttons.delete.ud_toggle_status }}, {{ post.thread_id }}, {{ post.id }}, '{{ post.author_user_id }}', '{{ post_date }}', '{{ csrf_token }}' )" title="{{ post_buttons.delete.ud_button_title }}" aria-label="{{ post_buttons.delete.ud_button_title }}"><i class="fa {{ post_buttons.delete.ud_button_icon }}"></i></a>
-                <a class="post_button edit-post-button" onClick="editPost({{ post.id }}, {{ post.thread_id }}, {{ post_buttons.edit.shouldEditThread }}, {{ render_markdown?1:0 }}, '{{ csrf_token }}')" title="{{ post_buttons.edit.edit_button_title }}" aria-label="{{ post_buttons.edit.edit_button_title }}"><i class="fas fa-edit"></i></a>
+                <a class="post_button delete-post-button" tabindex = "0" onClick="deletePostToggle({{ post_buttons.delete.ud_toggle_status }}, {{ post.thread_id }}, {{ post.id }}, '{{ post.author_user_id }}', '{{ post_date }}', '{{ csrf_token }}' )" title="{{ post_buttons.delete.ud_button_title }}"><i class="fa {{ post_buttons.delete.ud_button_icon }}" aria-hidden="true"></i></a>
```
Comment (elihschiff) [inline site/app/templates/forum/CreatePost.twig]: You should remove this aria-hidden

File: site/app/templates/forum/CreatePost.twig [hunk 7]
```diff
@@ -41,38 +41,38 @@
...
 
         {% if userGroup <= 2 and post.author_user_id != current_user %}
-            <a class="post-email-toggle" onClick='$(this).next().toggle();' title="Show/Hide email address" aria-label="Show/Hide email address"><i class="fas fa-envelope"></i></a>
+            <a class="post-email-toggle" tabindex = "0" onClick='$(this).next().toggle();' title="Show/Hide email address"><i class="fas fa-envelope" aria-hidden="true"></i></a>
             <a href="mailto:{{ author_email }}" style="display: none;">{{ author_email }}</a>
         {% endif %}
 
         {% if userGroup <= 2 %}
-            <a class="post-user-info" onClick='changeName(this.parentNode, {{ post_user_info.info_name }}, {{ post_user_info.visible_user_json}}, {{ post_user_info.jscriptAnonFix}})' title="Show full user information" aria-label="Show full user information"><i class="fas fa-eye"></i></a>
+            <a class="post-user-info" tabindex = "0" onClick='changeName(this.parentNode, {{ post_user_info.info_name }}, {{ post_user_info.visible_user_json}}, {{ post_user_info.jscriptAnonFix}})' title="Show full user information"><i class="fas fa-eye" aria-hidden="true"></i></a>
```
Comment (elihschiff) [inline site/app/templates/forum/CreatePost.twig]: You should remove this aria-hidden

File: site/app/templates/forum/CreatePost.twig [hunk 8]
```diff
@@ -41,38 +41,38 @@
...
     {% if includeReply and (userGroup <= 3 or post.author_user_id == current_user) and first and thread_resolve_state == -1 %}
-        <a class="btn btn-default btn-sm post_button_color text-decoration-none" onClick="changeThreadStatus({{ post.thread_id }})" title="Mark thread as resolved">Mark as resolved</a>
+        <a class="btn btn-default btn-sm post_button_color text-decoration-none" tabindex = "0" onClick="changeThreadStatus({{ post.thread_id }})" title="Mark thread as resolved">Mark as resolved</a>
     {% endif %}
 
     <span class="post-action-container">
 
         {% if userGroup <= 2 and post.author_user_id != current_user %}
-            <a class="post-email-toggle" onClick='$(this).next().toggle();' title="Show/Hide email address" aria-label="Show/Hide email address"><i class="fas fa-envelope"></i></a>
+            <a class="post-email-toggle" tabindex = "0" onClick='$(this).next().toggle();' title="Show/Hide email address"><i class="fas fa-envelope" aria-hidden="true"></i></a>
```
Comment (elihschiff) [inline site/app/templates/forum/CreatePost.twig]: You should remove this aria-hidden

Comment (elihschiff) [reply site/app/templates/forum/CreatePost.twig]:
```suggestion
            <a class="post-email-toggle" tabindex = "0" onClick='$(this).next().toggle();' title="Show/Hide email address" aria-label="Show/Hide email address"><i class="fas fa-envelope"></i></a>
```

File: site/app/templates/forum/CreatePost.twig (hunk 7 above)
Comment (elihschiff) [reply site/app/templates/forum/CreatePost.twig]:
```suggestion
            <a class="post-user-info" tabindex = "0" onClick='changeName(this.parentNode, {{ post_user_info.info_name }}, {{ post_user_info.visible_user_json}}, {{ post_user_info.jscriptAnonFix}})' title="Show full user information" aria-label="Show full user information"><i class="fas fa-eye"></i></a>
```

File: site/app/templates/forum/CreatePost.twig (hunk 5 above)
Comment (elihschiff) [reply site/app/templates/forum/CreatePost.twig]:
```suggestion
                <a class="post_button edit-post-button" tabindex = "0" onClick="editPost({{ post.id }}, {{ post.thread_id }}, {{ post_buttons.edit.shouldEditThread }}, {{ render_markdown?1:0 }}, '{{ csrf_token }}')" title="{{ post_buttons.edit.edit_button_title }}" aria-label="{{ post_buttons.edit.edit_button_title }}"><i class="fas fa-edit"></i></a>
```

File: site/app/templates/forum/CreatePost.twig (hunk 6 above)
Comment (elihschiff) [reply site/app/templates/forum/CreatePost.twig]:
```suggestion
                <a class="post_button delete-post-button" tabindex = "0" onClick="deletePostToggle({{ post_buttons.delete.ud_toggle_status }}, {{ post.thread_id }}, {{ post.id }}, '{{ post.author_user_id }}', '{{ post_date }}', '{{ csrf_token }}' )" title="{{ post_buttons.delete.ud_button_title }}" aria-label="{{ post_buttons.delete.ud_button_title }}"><i class="fa {{ post_buttons.delete.ud_button_icon }}"></i></a>
```

File: site/app/templates/forum/CreatePost.twig (hunk 8 above)
Comment (elihschiff) [reply site/app/templates/forum/CreatePost.twig]:
```suggestion
            <a class="post-email-toggle" tabindex = "0" onClick='$(this).next().toggle();' title="Show/Hide email address" aria-label="Show/Hide email address"><i class="fas fa-envelope"></i></a>
```

File: site/app/templates/forum/CreatePost.twig (hunk 7 above)
Comment (elihschiff) [reply site/app/templates/forum/CreatePost.twig]:
```suggestion
            <a class="post-user-info" tabindex = "0" onClick='changeName(this.parentNode, {{ post_user_info.info_name }}, {{ post_user_info.visible_user_json}}, {{ post_user_info.jscriptAnonFix}})' title="Show full user information" aria-label="Show full user information"><i class="fas fa-eye"></i></a>
```

File: site/app/views/forum/ForumThreadView.php [hunk 9]
```diff
@@ -207,8 +207,8 @@ public function showForumThreads($user, $posts, $unviewed_posts, $threadsHead, $
             array(
                 "required_rank" => 4,
                 "display_text" => 'Create Thread',
-                "style" => 'float:right;position:relative;top:3px;',
-                "link" => array(true, $this->core->buildCourseUrl(['forum', 'threads', 'new'])),
+                "style" => 'position:absolute;top:3px;right:0px',
+                 "link" => array(true, $this->core->buildCourseUrl(['forum', 'threads', 'new'])),
```
Comment (elihschiff) [inline site/app/views/forum/ForumThreadView.php]:
```suggestion
                "link" => array(true, $this->core->buildCourseUrl(['forum', 'threads', 'new'])),
```

File: site/app/templates/forum/CreatePost.twig (hunk 6 above)
Comment (jchulton) [reply site/app/templates/forum/CreatePost.twig]: Addressing in commit below

File: site/app/templates/forum/CreatePost.twig (hunk 5 above)
Comment (jchulton) [reply site/app/templates/forum/CreatePost.twig]: Addressing in commit below

File: site/app/templates/forum/CreatePost.twig [hunk 10]
```diff
@@ -41,38 +41,37 @@
...
     {% if includeReply and (userGroup <= 3 or post.author_user_id == current_user) and first and thread_resolve_state == -1 %}
-        <a class="btn btn-default btn-sm post_button_color text-decoration-none" onClick="changeThreadStatus({{ post.thread_id }})" title="Mark thread as resolved">Mark as resolved</a>
+        <a class="btn btn-default btn-sm post_button_color text-decoration-none" tabindex = "0" onClick="changeThreadStatus({{ post.thread_id }})" title="Mark thread as resolved">Mark as resolved</a>
     {% endif %}
 
     <span class="post-action-container">
 
         {% if userGroup <= 2 and post.author_user_id != current_user %}
-            <a class="post-email-toggle" onClick='$(this).next().toggle();' title="Show/Hide email address" aria-label="Show/Hide email address"><i class="fas fa-envelope"></i></a>
-            <a href="mailto:{{ author_email }}" style="display: none;">{{ author_email }}</a>
```
Comment (elihschiff) [inline site/app/templates/forum/CreatePost.twig]: did you mean to remove this line?

**Pull request merged**